[sindhurajain] iP

sindhurajain opened the pull request:
### myChats

> “Chatbots represent a new trend in how people access information, make decisions and communicate.” - Christie Pitts ([source](https://www.supplychaintoday.com/chatbot-quotes-by-top-minds/))

Do **you** want a chat bot that is:

- ~~Fun~~ Very fun
- Easy to use 
- Intuitive

And also...

- [ ] Helpful
- [ ] Reliable
- [ ] User-friendly

Download [myChats](https://github.com/sindhurajain/ip/) _today_! :star_struck:

Here are some features of myChats:

1. Add task
2. Delete task
3. Mark or unmark task as done
4. Search for task

To run the project, go to the `Duke` class and press the 'run' button at
```java
public static void main(String[] args) {
    Duke duke = new Duke("data/duke.txt");
    duke.startChat();
} 
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

sindhurajain committed:
Implemented Level 0. Rename, Greet, Exit

sindhurajain committed:
javadocs for Level-0

sindhurajain committed:
Implemented Level 1 and javadoc comments

sindhurajain committed:
Implemented Level-2 and javadoc comments

sindhurajain committed:
Implemented Level-3 and javadoc comments

sindhurajain committed:
Implemented Level 4 but yet to write javadoc comments

sindhurajain committed:
Wrote javadoc comments for Level 4.

sindhurajain committed:
Implemented A-TextUiTesting. Test result: PASSED

sindhurajain committed:
Implemented Level-5 with the 2 minimal errors and additional errors handled

sindhurajain committed:
Wrote javadoc comments for Level-5.

sindhurajain committed:
Implemented Level-6. Yet to write javadoc comments and change from primitive array to ArrayList.

sindhurajain committed:
Wrote javadoc comments for Level-6.

sindhurajain committed:
Level-6: converted primitive array to ArrayList

sindhurajain committed:
Implement Level-7

sindhurajain committed:
Merge branch branch-Level-7 into master

sindhurajain committed:
Implement Level-8

sindhurajain committed:
Merge branch 'branch-Level-8' into master

sindhurajain committed:
Implement A-MoreOOP

sindhurajain committed:
Implement A-Packages

sindhurajain committed:
Merge branch 'add-gradle-support' of https://github.com/sindhurajain/ip into gradle2

sindhurajain committed:
Merge branch 'gradle2'

sindhurajain committed:
Implement A-Gradle

sindhurajain committed:
Add JUnit tests for Parser, Deadline, Event, Todo

sindhurajain committed:
Add JUnit tests for TaskList

sindhurajain committed:
Test parallel branch

sindhurajain committed:
Write Storage javadoc comments

sindhurajain committed:
Write comments for find function

sindhurajain committed:
Write coding standard comments

sindhurajain committed:
Implement Level-9

sindhurajain committed:
Write javadoc comments for Command.java, DeadlineCommand.java and DeleteCommand.java

sindhurajain committed:
Write Javadoc comments fr EventCommand, ExitCommand and ListCommand

sindhurajain committed:
Write Javadoc comments for MarkCommand, OnCommand and ToDoCommand

sindhurajain committed:
Add javadoc comments to UnMarkCommand.java, modify javadoc comments of MarkCommand.java

sindhurajain committed:
Write javadoc comments for DukeException.java and Duke.java

sindhurajain committed:
Write javadoc comments for Parser.java

sindhurajain committed:
Write javadoc comments for Storage.java, TaskList.java and Ui.java

sindhurajain committed:
Write javadoc comments for Deadline.java, Event.java, Task.java and Todo.java

sindhurajain committed:
Implement A-JavaDoc

sindhurajain committed:
Tweak the code to comply with Java coding standard

sindhurajain committed:
Merge branch 'branch-A-CodingStandard' into branch-A-JavaDoc

# Conflicts:
#	src/main/java/duke/command/DeadlineCommand.java
#	src/main/java/duke/command/ExitCommand.java
#	src/main/java/duke/command/ListCommand.java
#	src/main/java/duke/command/OnCommand.java
#	src/main/java/duke/command/UnMarkCommand.java
#	src/main/java/duke/main/Duke.java
#	src/main/java/duke/main/Parser.java
#	src/main/java/duke/main/Storage.java
#	src/main/java/duke/main/TaskList.java
#	src/main/java/duke/main/Ui.java
#	src/main/java/duke/task/Deadline.java
#	src/main/java/duke/task/Todo.java

sindhurajain committed:
Merge branch 'branch-Level-9' into branch-A-JavaDoc

# Conflicts:
#	src/main/java/duke/main/Parser.java
#	src/main/java/duke/main/Ui.java
#	src/main/java/duke/task/Task.java

sindhurajain committed:
Merge branch 'branch-A-JavaDoc'

sindhurajain committed:
Add javadoc comments and modify code style for Level-9

sindhurajain committed:
Modify javadoc comments

sindhurajain committed:
Implement Level-10

sindhurajain committed:
Adjust spacing and round image borders

sindhurajain committed:
Write javadoc comments and adhere to Java Coding Standard

sindhurajain committed:
Merge branch 'branch-Level-10'

sindhurajain committed:
Modify javadoc comments, adjust spacing and colour in GUI

sindhurajain committed:
Modify build.gradle to create new JAR file

appleraincoat commented on src/main/java/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,44 @@
+public class DeadlineCommand extends Command {
+
+    private String description;
+    private String by;
```
Perhaps you can consider renaming "by" to "byWhen"?

appleraincoat commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,46 @@
+ */
+public class DeleteCommand extends Command {
+
+    private int zeroItem;
```
It seems like this variable is representing an index. Perhaps you can consider giving a clearer name, such as "indexToDelete"?

appleraincoat commented on src/main/java/duke/command/EventCommand.java:
```diff
@@ -0,0 +1,48 @@
+
+    private String description;
+    private String startTime;
+    private String endTime;
```
I like the choice of variable names here!

appleraincoat commented on src/main/java/duke/command/ExitCommand.java:
```diff
@@ -0,0 +1,32 @@
+     * @return True as ExitCommand is an exit command.
+     */
+    @Override
+    public boolean isExit() {
```
I like how you give boolean methods names that sound boolean too, nice!

appleraincoat commented on src/main/java/duke/main/Parser.java:
```diff
@@ -0,0 +1,171 @@
+package duke.main;
```
Great choice of method names for this class!

appleraincoat commented on src/main/java/duke/main/TaskList.java:
```diff
@@ -0,0 +1,84 @@
+     * @param num The zero-indexed task index to mark as not done.
+     */
+     public void unMarkAsDone(int num) {
+        tasks.get(num).unMarkAsDone();
```
Perhaps you can consider renaming these 2 methods to just markTask and unmarkTask?

appleraincoat commented on src/main/java/duke/main/Ui.java:
```diff
@@ -0,0 +1,163 @@
+     * @param tasks ArrayList of tasks to search for tasks
+     * that occur on the given target date.
+     */
+    public void displayTasksOn(LocalDate targetDate, ArrayList<Task> tasks) {
```
Just a small nitpick: displayTasksOnDate may be a slightly clearer name

appleraincoat reviewed (commented):
Besides some small nits, I think overall you did a good job on the naming! Code style looks good to me too!

sindhurajain committed:
Adjust spacing in code

sindhurajain committed:
Implement A-Assertions Increment

Current code does not document important assumptions that tasks should
not be null and task size should be more than or equal to 0.

Let's,
* create the assertion in the TaskList class that task size should be
more than or equal to 0
* create the assertion in the Ui class and TaskList class that tasks
should not be null

sindhurajain committed:
Implement A-CodeQuality

Some code does not adhere to the code quality guidelines.

Code that adheres to the code quality guidelines is more readable and
understandable.

Let's,
* make the happy path prominent in the Duke class
* avoid complicated boolean expressions in the DeleteCommand, Parser,
TaskList, MarkCommand and UnMarkCommand classes by evaluating boolean
expressions in steps

sindhurajain committed:
Merge pull request #3 from sindhurajain/branch-A-Assertions

Implement A-Assertions Increment

sindhurajain committed:
Merge branch 'master' into branch-A-CodeQuality

sindhurajain committed:
Merge pull request #4 from sindhurajain/branch-A-CodeQuality

Implement A-CodeQuality

sindhurajain committed:
Merge branch 'branch-A-CodeQuality' of https://github.com/sindhurajain/ip into branch-A-CodeQuality

sindhurajain committed:
Implement B-ViewSchedules

Current implementation allows users to view their schedule on a
specific date using the "on" command.

This features provides a way for users to view tasks on a specific date
in the form of a schedule.

Let's,
* modify the OnCommand class and rename it to ViewCommand,
refactoring the current code
* modify the Parser class such that the "view" command displays the
scheduled tasks on the given date instead of the "on" command
* modify the JUnit test cases in the ParserTest class to account for the
modified "view" command and the refactored ViewCommand class

sindhurajain committed:
Implement Undo command

Current code does not allow users to undo their most recently added
task.

The undo command allows users to undo their most recently added task by
entering "undo".

Let's,
* parse the "undo" command in the Parser class
* add a method in the TaskList class to undo the command
* add a method in the Ui class to display the tasks after the undo
command is performed
* rename the previous taskResponse method, which displays the list
after a task is added, to addResponse for clarity
* add an UndoCommand class to process the undo command by calling the
methods in the other classes to undo the task

sindhurajain committed:
Merge pull request #5 from sindhurajain/branch-B-ViewSchedules

Implement B-ViewSchedules

sindhurajain committed:
Merge pull request #6 from sindhurajain/branch-C-Undo

Implement Undo command

sindhurajain committed:
Merge pull request #7 from sindhurajain/branch-A-CodeQuality

Merge branch 'branch-A-CodeQuality' of https://github.com/sindhurajai…

sindhurajain committed:
Refactor code to use the name MyChats instead of Duke and split long
method in Parser class into multiple methods

sindhurajain committed:
Refactor code to use the name MyChats instead of Duke and split long
method in Parser class into multiple methods

sindhurajain committed:
Set title in GUI

sindhurajain committed:
Adjust spacing in GUI

sindhurajain committed:
Adjust title and spacing of GUI

sindhurajain committed:
Update README

sindhurajain committed:
Rename mark and unmark command

sindhurajain committed:
Rename indexed variable names

sindhurajain committed:
Modify build.gradle and chatbot icons

sindhurajain committed:
Modify screenshot in user guide
